idx/memory/memory.go

```diff
@@ -1192,7 +1220,9 @@ func (m *UnpartitionedMemoryIdx) Delete(orgId uint32, pattern string) ([]idx.Arc
 
 	statMetricsActive.Set(len(m.defById))
 	statDeleteDuration.Value(time.Since(pre))
-
+	if len(deletedDefs) > 0 {
+		m.findCache.Purge(orgId)
```

Comment:
if `len(deletedDefs)` is small, it may make more sense to do a sequence of `InvalidateFor` calls instead of Purge ?

Reply:
updated